packages/live-share-inking/src/core/SharedInkingSession.ts

```diff
@@ -0,0 +1,695 @@
+                : new BuiltInCursorVisual(clientId, userInfo);
+
+            this._cursorVisualsMap.set(clientId, cursorVisual);
+        }
+
+        if (!this._cursorVisualsHost.contains(cursorVisual.renderedElement)) {
+            this._cursorVisualsHost.appendChild(cursorVisual.renderedElement);
+        }
+
+        return cursorVisual;
+    }
+
+    private removeCursor(clientId: string) {
+        const cursorVisual = this._cursorVisualsMap.get(clientId);
+
+        if (cursorVisual) {
+            if (this._cursorVisualsHost.contains(cursorVisual.renderedElement)) {
+                this._cursorVisualsHost.removeChild(cursorVisual.renderedElement);
+            }
+        }
+
+        // We do not remove the cursor component from the cursor map so that
+        // we don't lose the color it got automatically attributed.
+    }
+
+    private updateCursorPosition(clientId: string, userInfo?: IUserInfo, position?: IPoint) {
+        if (this._inkingManager) {
+            if (position) {
+                const cursorVisual = this.getCursor(clientId, userInfo);
+
+                const screenPosition = this._inkingManager.viewportToScreen(position);
+
+                cursorVisual.setPosition(screenPosition);
+            }
+            else {
+                this.removeCursor(clientId);
+            }
+        }
+    }
+
+    protected async initializingFirstTime(): Promise<void> {
+        this._dryInkMap = SharedMap.create(this.runtime, SharedInkingSession.dryInkMapKey);
+
+        this.root.set(SharedInkingSession.dryInkMapKey, this._dryInkMap.handle);
+    }
+
+    protected async hasInitialized(): Promise<void> {
+        const handle = this.root.get<IFluidHandle<SharedMap>>(SharedInkingSession.dryInkMapKey);
+
+        if (handle) {
+            this._dryInkMap = await handle.get();
+        }
+        else {
+            throw new Error(`Unable to get SharedMap with key "${SharedInkingSession.dryInkMapKey}"`);
+        }
+    }
+
+    /**
+     * Optional callback that allows the consuming application to provide a
+     * friendly display name and/or a picture that will be used on remote devices
+     * to render shared cursors.
+     */
+    onGetLocalUserInfo?: () => IUserInfo | undefined;
+
+    /**
+     * Optional callback that allows the consuming application to provide its own
+     * cursor visual implementation by extending the abstract `CursorVisual` class.
+     * The callback is passed the cursor info retrieved via the `onGetCursorInfo`
+     * calback, if provided.
+     */
+    onCreateCursorVisual?: (clientId: string, userInfo?: IUserInfo) => CursorVisual;
+
+    /**
+     * Starts the live inking session.
+     * @param hostElement The element to attach the InkingManager to.
+     * @returns An InkingManager instance that can be used by the applications
+     * to set the tool, brush, add strokes and more.
+     */
+    synchronize(hostElement: HTMLElement): InkingManager {
```

Comment:
>synchronize [](http://example.com/codeflow?start=4&length=11)

While this is consistent with the `EphemeralMediaSession` we actually use a separate `initialize()` method to "start" the session. The session can be initialized only once but you can change the media play you synchronize with as many times as you'd like.

I don't know if the same semantics apply here but worth talking through.

Reply:
@stevenic you and Ryan need to get on the same page here :) I originally had it as `initialize` and moved to `synchronize` because Ryan told me it was more consistent...

Reply:
Switched back to `initialize`:
```typescript
async SharedInkingSession.initialize(inkingManager: InkingManager);
```